packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -168,7 +190,26 @@ export class AssociationManager {
     return `- Portfolio: ${portfolio.node.path} | Product: ${product.node.path}`;
   }
 
+  private static formatAssertions(scope: cdk.Resource, assertions : Assertion[]): {[key: string]: string}[] {
+    const formattedAssertions: {[key: string]: string}[] = [];
+    assertions.forEach(assertion => formattedAssertions.push({
+      Assert: scope.stack.resolve(assertion.assert),
+      ...(assertion.assertDescription) && { AssertDescription: assertion.assertDescription },
```

Comment:
```suggestion
      AssertDescription: assertion.assertDescription,
```

Reply:
description should actually be optional, is this a fair way to cast it? AssertDescription: assertion.description as string . Get's around the undefined type and won't create the key:value pair if it's undefined.

Reply:
`assertDescription` is optional, so that will result in `AssertDescription: undefined`, which should be skipped by `JSON.stringify()` (a unit test will confirm).

Reply:
okay, but since `JSON.stringify()` is called by the parent function I need to update this type to be `{[key: string]: string | undefined }` , right?

Reply:
Sure. I don't think that's a problem - that seems to reflect the actual type better than `{ [key: string]: string }`.